demo/lightning/image-segmentation/data.py

```diff
@@ -1,23 +1,144 @@
-import torch
+"""
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+
 import os
 import lightning.pytorch as pl
+import random
+import numpy as np
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+from torchvision import transforms
 
 from dataflux_pytorch import dataflux_mapstyle_dataset
+from pytorch_loader import DatafluxPytTrain
 
 
 class Unet3DDataModule(pl.LighitningDataModule):
-    def __init__(self, gcs_bucket, prefix):
-        self.data_dir = os.path.join(gcs_bucket, prefix)
+
+    def __init__(self, args):
+        super().__init__()
+        self.args = args
     
     def prepare_data(self):
         pass
 
-    def setup(self, state="train"):
-        # Init DatafluxPytTrain object 
-        # Pass transform function as argument - modify DatafluxPyTrain constructor
-        pass
+    def setup(self, stage="fit"):
+        if stage == "fit":
+            train_data_kwargs = {
+                "patch_size": self.args.input_shape,
+                "oversampling": self.args.oversampling,
+                "seed": self.args.seed,
+                "images_prefix": self.args.images_prefix,
+                "labels_prefix": self.args.labels_prefix,
+                "transforms": get_train_transforms(),
+                }
+            self.train_dataset = DatafluxPytTrain(
+                project_name=self.args.gcp_project,
+                bucket_name=self.args.gcs_bucket,
+                config=dataflux_mapstyle_dataset.Config(sort_listing_results=True),
+                **train_data_kwargs,
+                )
+            self.train_sampler = None
+            if self.args.num_workers > 1:
+                self.train_sampler = DistributedSampler(
+                    self.train_dataset, 
+                    seed=self.args.seed, 
+                    drop_last=True)
 
     def train_dataloader(self):
-        # Init dataflux dataloader
-        # Wrap it in torch.DataLoader
-        pass
+        return  DataLoader(
```

Comment:
nit: There is a double space here. Have you run the formatter? If not, you should be able to install it from the Extensions > Recommended Extensions panel on VS Code and then it will run on save